### ISSUE TITLE
Settings: why do we have custom place for actions and not use editor secondary actions

### ISSUE THREAD

**Issue by isidorn** [Microsoft/vscode#61590]
@sandy081 and me find it weird that we have a custom place in settings where we keep secondary actions and we do not use editor secondary actions. Why is that, what is the reasoning behind this design choice?
This way we are inconsistent with out other settings experiences, for example Keybinginds Editor.

fyi @misolori 

![screen shot 2018-10-23 at 12 08 11](https://user-images.githubusercontent.com/1926584/47353830-7848cc80-d6bd-11e8-9202-2427fa007432.png)

**Comment by roblourens** [comment]
I think we were concerned that the editor gutter actions were too far away. The settings editor is different because its content is centered  so there's no content near the common actions, like your screenshot shows. But I'm thinking I should just use that spot anyway.

**Comment by isidorn** [comment]
I would go with consistency here. Especially since these actions are not for novice users but for expeirenced users who are aware of editor primary and secondary actions.

**Comment by misolori** [comment]
The other reason we decided to not use the editor actions was that majority of the actions modify your window (markdown preview, diff, changes, etc.) and Settings had filters that are more contextual and are similar to the extension search + context menu.

**Comment by misolori** [comment]
I agree that open settings.json is similar to those other views.

We can definitely grab numbers to see how many people are accessing the json via the current dropdown, but I think a more appropriate way to measure that is to do a short usability study, since this will also apply to the keybindings page. Maybe @stevencl has thoughts on how long it would take to get feedback from users.

**Comment by misolori** [comment]
I think we're in agreement that the experiences should be the same/consistent. Here are the two different methods, @roblourens @isidorn @sandy081 let me know if this aligns with your thinking:

## Use editor actions
This aligns with the current keybindings page and moves the "Open settings.json" into an action/icon and adds the filters to the editor context menu.

![image](https://user-images.githubusercontent.com/35271042/47373221-36f5f400-d6a0-11e8-9781-f852acc0c318.png)


## Use contextual actions
This is how the current settings page is except we'd still keep the "Open settings.json" in the editor actions.

![image](https://user-images.githubusercontent.com/35271042/47373351-82a89d80-d6a0-11e8-8037-2fc1f7a69e8d.png)

**Comment by sandy081** [comment]
@misolori I liked the first one that uses editor actions. This will make it consistent with other editors too. Show two `...` looks to me redundant and not good.

Regarding opening `json` file action, if we show it as primary action then we should need good icon to identify the action. Otherwise we can move it as secondary action similar to how it is now in Settings editor but as editor actions.

**Comment by stevencl** [comment]
Is a study required?  Seems like there is general consensus that consistency is preferred?

**Comment by misolori** [comment]
@stevencl doesn't look like we'll need a study, we'll just monitor the feedback with this change.

**Comment by isidorn** [comment]
@misolori thanks for skethiching that up. Curly bracket icon might just work.
It would be great if we could get this into insiders so we get feedback and potnetialy change before we release in 10 days.

**Comment by roblourens** [comment]
The keybindings editor uses the file/arrow 
![image](https://user-images.githubusercontent.com/323878/47447847-6aee1980-d773-11e8-828c-aefcf988c558.png)

I like the braces too. Chrome devtools uses that icon for "Pretty print", I don't really know why.

**Comment by misolori** [comment]
We also use that icon in the Markdown preview as the "Go to source" and think we can use either. However, in #61587 there was discussion that the file/arrow icon was hard to notice and often missed.

**Comment by isidorn** [comment]
File/arrow icon was unnoticable by everybody in zurich that tried to find this functionality - it associates me to the "git get me out of diff view", since we use it there already.

**Comment by roblourens** [comment]
I can do this if you send me the braces icon @misolori ?

**Comment by sandy081** [comment]
I am planning this to next milestone so that we can run with these actions in insiders to see if there are no complains.

**Comment by misolori** [comment]
@roblourens sent you the icons

**Comment by roblourens** [comment]
Sounds good @sandy081

**Comment by isidorn** [comment]
The hover titles can still be aligned @roblourens and @sandy081 
Though the overall alignment is good adding verified

**Comment by sandy081** [comment]
I updated Keyboard shortcuts actions title to `Open Keyboard Shortcuts` and `Open Keyboard Shortcuts (JSON)`. Did not add `(UI)` part as `Open Keyboard Shortcuts` is the default command to open the editor.

@roblourens Same can be done for Settings editor actions?

**Comment by usernamehw** [comment]
@sandy081, can those items **not** be named that way (in parenthes). vscode search not matching them: #27636 and there is no progress on that issue. It would be ideal for the time being not using brackets at all.

**Comment by roblourens** [comment]
@sandy081 looks like they match now so there's nothing to do for settings right?

The filtering issue is annoying...

**Comment by sandy081** [comment]
Yeah they match now, but I think Settings has another action called Open Settings (UI).

**Comment by roblourens** [comment]
Now there are two different actions with the same name, I'm reverting that commit. Settings has 3 - open the configured editor, open the settings ui, or open the settings json.
